test/sql/join/semianti/antijoin.test

```diff
@@ -0,0 +1,105 @@
+# name: test/sql/join/semianti/antijoin.test
+# description: Test positional joins
+# group: [semianti]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE left_table (a INTEGER, b INTEGER, c INTEGER);
+
+statement ok
+INSERT INTO left_table VALUES(42, 1, 1), (43, 1, 1);
+
+statement ok
+CREATE TABLE right_table (a INTEGER, b INTEGER);
+
+statement ok
+INSERT INTO right_table VALUES(42, 1);
+
+# * expression expands all of the left table
+query III
+SELECT * FROM left_table ANTI JOIN right_table ON left_table.a = right_table.a;
```

Comment:
Could we add tests with:

* SEMI/ANTI join inside correlated subqueries (so that the left/right side contains correlated columns)
* Joins with complex predicates (e.g. `tbl1.a + tbl2.a = 42 OR tbl1.b + tbl2.b = 42`)
* Inequality joins (e.g. `tbl1.a <> tbl2.a`)
* Range joins (e.g. `tbl1.a > tbl2.a`)